kymatio/backend/numpy_backend.py

```diff
@@ -1,73 +1,81 @@
-import numpy as np
 
-def input_checks(x):
-    if x is None:
-        raise TypeError('The input should be not empty.')
-
-def complex_check(x):
-    if not _is_complex(x):
-        raise TypeError('The input should be complex.')
-
-def real_check(x):
-    if not _is_real(x):
-        raise TypeError('The input should be real.')
-
-
-def modulus(x):
-    """
-        This function implements a modulus transform for complex numbers.
-
-        Usage
-        -----
-        x_mod = modulus(x)
-
-        Parameters
-        ---------
-        x: input complex tensor.
-
-        Returns
-        -------
-        output: a real tensor equal to the modulus of x.
-
-    """
-    return np.abs(x)
-
-
-def _is_complex(x):
-    return (x.dtype == np.complex64) or (x.dtype == np.complex128)
-
-
-def _is_real(x):
-    return (x.dtype == np.float32) or (x.dtype == np.float64)
-
-
-def cdgmm(A, B):
-    """
-        Complex pointwise multiplication between (batched) tensor A and tensor B.
-
-        Parameters
-        ----------
-        A : tensor
-            A is a complex tensor of size (B, C, M, N, 2)
-        B : tensor
-            B is a complex tensor of size (M, N) or real tensor of (M, N)
-
-        Returns
-        -------
-        C : tensor
-            output tensor of size (B, C, M, N, 2) such that:
-            C[b, c, m, n, :] = A[b, c, m, n, :] * B[m, n, :]
-
-    """
-
-    if not _is_complex(A):
-        raise TypeError('The first input must be complex.')
-
-    if A.shape[-len(B.shape):] != B.shape[:]:
-        raise RuntimeError('The inputs are not compatible for '
-                           'multiplication.')
-
-    if not _is_complex(B) and not _is_real(B):
-        raise TypeError('The second input must be complex or real.')
-
-    return A * B
+   
+class NumpyBackend:
+    def __init__(self, name='numpy'):
+        import numpy
```

Comment:
I believe it shouldn't trigger a numpy import unless we call the init functions. 

I thought the best idea was to import it inline, I believe that might have been what @janden wanted.

Reply:
Yeah I think nothing is really stopping us from using a global import.

I'd generally also like to get away as much as possible from making an instance. Ideally all functions are static on the class and we use inheritance to pass down those static functions

Reply:
> Yeah I think nothing is really stopping us from using a global import.

Agree. That's how it is for the other backends, as far as I can tell.

> I'd generally also like to get away as much as possible from making an instance. Ideally all functions are static on the class and we use inheritance to pass down those static functions

I remember inheriting static functions as being slightly tricky to deal with when it comes to inheritance, but this may be a false memory. Overall, I think it's simple enough to have these singleton instances since it allows us to store some internal state (like these module references).

Reply:
why can't we just write those into the class?
I'm happy to keep any state that is necessary, but would avoid it if it's easy